src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -1,45 +1,55 @@
 package com.gu.fastly
 
-import org.apache.commons.codec.digest.DigestUtils
-import com.amazonaws.services.kinesis.clientlibrary.types.UserRecord
+import java.io.IOException
+
 import com.amazonaws.services.kinesis.model.Record
+import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
 import com.amazonaws.services.lambda.runtime.events.KinesisEvent
+import com.gu.crier.model.event.v1.{Event, EventPayload, EventType}
+import com.gu.fastly.CrierEventProcessor.{decodeRecord, successfulEvents}
 import okhttp3._
-import com.gu.crier.model.event.v1._
+import org.apache.commons.codec.digest.DigestUtils
+
 import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.{ Future, Promise, Await, duration }
 
-class Lambda {
+class Lambda extends RequestHandler[KinesisEvent, Unit] {
 
   private val config = Config.load()
   private val httpClient = new OkHttpClient()
 
-  def handle(event: KinesisEvent) {
+  def handleRequest(event: KinesisEvent, context: Context): Unit = {
+    Await.result(processEvent(event), duration.Duration(context.getRemainingTimeInMillis, duration.MILLISECONDS))
+    println(s"Finished.")
+  }
+
+  def processEvent(event: KinesisEvent): Future[Unit] = {
     val rawRecords: List[Record] = event.getRecords.asScala.map(_.getKinesis).toList
-    val userRecords = UserRecord.deaggregate(rawRecords.asJava)
-
-    println(s"Processing ${userRecords.size} records ...")
-
-    CrierEventProcessor.process(userRecords.asScala) { event =>
-      (event.itemType, event.eventType) match {
-        case (ItemType.Content, EventType.Delete) =>
-          sendFastlyPurgeRequestAndAmpPingRequest(event.payloadId, Hard, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey)
-        case (ItemType.Content, EventType.Update) =>
-          sendFastlyPurgeRequest(event.payloadId, Soft, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey)
-          sendFastlyPurgeRequest(s"${event.payloadId}.json", Soft, config.fastlyApiNextgenServiceId, makeDotcomSurrogateKey(s"${event.payloadId}.json"), config.fastlyDotcomApiKey)
-          sendFastlyPurgeRequest(s"${event.payloadId}.json", Soft, config.fastlyMapiServiceId, makeMapiSurrogateKey(s"${event.payloadId}.json"), config.fastlyMapiApiKey)
```

Comment:
Why do we need the .json for Update events, but not RetrievableUpdate events? (TODO: check event structure)

Reply:
I think this was simply missed in #14. A  `RetrievableUpdate` is similar to `Update` for a business perspective. They only differ because there is a limit size on the payload one can send to Kinesis, so a `RetrievableUpdate` is `Update` with a payload that need to be retrieved.